src/main/java/org/hyperledger/fabric/sdk/BlockDeserializer.java

```diff
@@ -81,8 +79,12 @@ EnvelopeDeserializer getData(int index) throws InvalidProtocolBufferException {
     }
 
     byte[] getTransActionsMetaData() {
-
-        return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+        List<ByteString> metadataList = block.getMetadata().getMetadataList();
+        if (metadataList != null && metadataList.size() > Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE) {
```

Comment:
`block.getMetadata().getMetadataList()` always returns a List; never null. Getting the list of fine if you are going to work with that list (which you don't on the line below). You could also just work with the BlockMetadata and check `metadata.getMetadataCount()` before calling `metadata.getMetadata()`.